### DISCUSSION TITLE
Backend code using pocketbase hosting

### DISCUSSION THREAD

**Discussion by ajayyy123** [pockethost/pockethost#231] [Q&A]
How to write backend hooks or any code customization like I have written in the backend only verified users are allowed to login how to that here use pocketbase hosting.

**Comment by benallfree** [comment]
@ajayyy123 Welcome to PocketHost :)

> I have written in the backend

Can you explain more? Where have you written this?

**Comment by ajayyy123** [comment]
Hi Ben ,

Thanks for your reply

I was referring to backend code like this its for user is only allowed to
login after verification

func main() {
    app := pocketbase.New()

    app.OnRecordAuthRequest().Add(func(e *core.RecordAuthEvent) error {
        if !e.Record.Verified() {
            return apis.NewForbiddenError("You need to verify your account
first", nil)
        }

        return nil
    })
    if err := app.Start(); err != nil {
        log.Fatal(err)
    }

}

**Comment by benallfree** [comment]
@ajayyy123 I see now. PocketHost does not allow custom builds of PocketBase. As an alternative, you could consider using collection API rules to prevent unverified users from interacting with the data.

**Comment by ajayyy123** [comment]
How about writing any custom logic like any hooks

**Reply by benallfree** [reply]
We have experimental support for Deno-based cloud functions that use the same PocketBase JS SDK as your frontend would use. Using that SDK, you can listen for realtime change events and respond to them. PocketBase does not yet support webhooks though.

**Reply by ajayyy123** [reply]
I want to use  Deno-based cloud functions if you can provide me the guidance or any docs ...Its required in our project.

**Comment by ajayyy123** [comment]
Is there any option available to check using pb on the front if user mail is verified or not ....before login.

**Reply by benallfree** [reply]
Yes, you can use the PocketBase JS SDK to authenticate a user and then check their `verified` status. We do that in PocketHost: https://github.com/benallfree/pockethost/blob/0a6301c7c25a28df88d20361ea2368896f54679a/packages/pockethost.io/src/components/VerifyAccountBar.svelte#L32

**Reply by ajayyy123** [reply]
Thanks got it.

**Reply by ajayyy123** [reply]
Is there any problem in pockethost , all of sudden I cannt see my instances ..and not able to login..

**Reply by benallfree** [reply]
@ajayyy123 Please use #223 if you have more issues where you think PocketHost might be having trouble.

**Comment by benallfree** [comment]
@ajayyy123 https://github.com/pocketbase/pocketbase/discussions/2776#discussioncomment-6258531 will solve your problem.
